distributor-node/src/commands/operator/set-metadata.ts

```diff
@@ -31,18 +43,58 @@ export default class OperatorSetMetadata extends AccountsCommandBase {
   }
 
   async run(): Promise<void> {
-    const { bucketId, workerId, input, endpoint } = this.parse(OperatorSetMetadata).flags
+    const { bucketId, workerId, input, endpoint, operationalStatus: statusType } = this.parse(OperatorSetMetadata).flags
     const workerKey = await this.getDistributorWorkerRoleKey(workerId)
 
+    let operationalStatus: INodeOperationalStatusMetadata = {}
+    switch (statusType) {
+      case 'Normal': {
+        operationalStatus = { status: NodeOperationalStatusMetadata.OperationalStatus.NORMAL }
+        break
+      }
+      case 'NoService': {
+        operationalStatus = { status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE }
+        break
+      }
+      case 'NoServiceFrom': {
+        operationalStatus = {
+          status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+          noServiceFrom: (await this.datePrompt({ message: 'Enter No Service period start date' })).toISOString(),
+        }
+        break
+      }
+      case 'NoServiceDuring': {
+        operationalStatus = {
+          status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+          noServiceFrom: (await this.datePrompt({ message: 'Enter No Service period start date' })).toISOString(),
+          noServiceTo: (await this.datePrompt({ message: 'Enter No Service period end date' })).toISOString(),
+        }
+      }
+    }
+
     const validation = new ValidationService()
-    const metadata: IDistributionBucketOperatorMetadata = input
-      ? validation.validate('OperatorMetadata', JSON.parse(fs.readFileSync(input).toString()))
-      : { endpoint }
+    let metadata: IDistributionBucketOperatorMetadata
+    if (input) {
+      const params = validation.validate('OperatorMetadata', JSON.parse(fs.readFileSync(input).toString()))
+      metadata = {
+        ...params,
+        operationalStatus: {
+          ...params.operationalStatus,
+          status:
+            params.operationalStatus?.status === 'Normal'
+              ? NodeOperationalStatusMetadata.OperationalStatus.NORMAL
+              : NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+        },
```

Comment:
I think if somebody will try to set metadata without operational status then this will set it to `NO_SERVICE`?

Reply:
Addressed in [6ff4531](https://github.com/Joystream/joystream/pull/4793/commits/6ff45319432ff711ce35e281f01b0fba889089a3)